nativeunwind/elfunwindinfo/elfehframelookup.go

```diff
@@ -0,0 +1,51 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package elfunwindinfo // import "go.opentelemetry.io/ebpf-profiler/nativeunwind/elfunwindinfo"
+
+import (
+	"errors"
+	"sort"
+
+	"go.opentelemetry.io/ebpf-profiler/libpf/pfelf"
+)
+
+type FDE struct {
+	PCBegin uintptr
+	PCRange uintptr
+}
+
+// LookupFDE performs a binary search in .eh_frame_hdr for an FDE covering the given addr.
+func LookupFDE(ef *pfelf.File, addr uintptr) (FDE, error) {
```

Comment:
As this will be the API other packages can leverage, we should use `libpf.Address` as it is the commonly used type.
```suggestion
func LookupFDE(ef *pfelf.File, addr libpf.Address) (FDE, error) {
```